### PR TITLE
Adding BFS Graph Algorithm in Python 2

### LINKED ISSUE #458: Breadth First Search in Python2
This is an open issue for the Breadth-First Search algorithm implementation in python2. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by vale-c**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #458 [**](https://github.com/FOSS-UCSC/FOSSALGO/issues/458)

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [X] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [X] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [X] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [X] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [X] This pull request shall only be merged once the build passes.

-   [X] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex 
+        # If a adjacent has not been visited, then mark it 
+        # visited and enqueue it
+        for v in graph[current_vertex]:
+            if v not in visited:
+                visited.add(v)
+                queue.appendleft(v)
+
```
Comment (Danushka96) [inline algorithms/gr-bfsrh/python2/bfs.py:27]: Hi @vale-c 
Shall we add an example main function to show how someone can use this script?

Comment (vale-c) [reply algorithms/gr-bfsrh/python2/bfs.py:27]: Sounds like a good idea @Danushka96 , I will add a main function with an example!

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from collections import deque
+
+# graph: List[List[int]]
+# s: start vertex
+def bfs(graph, s):
+    # set is used to mark visited vertices
+    visited = set()
+
+    # create a queue for BFS
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        # Get all adjacent vertices of current_vertex 
+        # If a adjacent has not been visited, then mark it 
+        # visited and enqueue it
+        for v in graph[current_vertex]:
+          if graph[current_vertex] == None:
+            return
+            if v not in visited:
+                visited.add(v)
+                queue.appendleft(v)
+
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:29]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        current_vertex = queue.pop()
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex 
+        # If a adjacent has not been visited, then mark it 
+        # visited and enqueue it
+        for v in graph[current_vertex]:
+          if graph[current_vertex] == None:
+            return
+            if v not in visited:
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:26]: Codacy found an issue: [Unreachable code](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    # Mark the start vertex as visited and enqueue it
+    visited.add(s)
+    queue.appendleft(s)
+
+    while queue:
+        current_vertex = queue.pop()
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:20]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    # Mark the start vertex as visited and enqueue it
+    visited.add(s)
+    queue.appendleft(s)
+
+    while queue:
+        current_vertex = queue.pop()
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex 
+        # If a adjacent has not been visited, then mark it
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:21]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py (hunk 5 above)
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:20]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:29]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py (hunk 2 above)
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py (hunk 6 above)
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:21]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py (hunk 4 above)
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:26]: Codacy found an issue: [Unreachable code](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+from collections import deque
+
+# graph: List[List[int]]
+# s: start vertex
+def bfs(graph, s):
+    # set is used to mark visited vertices
+    visited = set()
+
+    # create a queue for BFS
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    # Mark the start vertex as visited and enqueue it
+    visited.add(s)
+    queue.appendleft(s)
+
+    while queue:
+        current_vertex = queue.pop()
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:20]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

File: algorithms/gr-bfsrh/python2/bfs.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    # Mark the start vertex as visited and enqueue it
+    visited.add(s)
+    queue.appendleft(s)
+
+    while queue:
+        current_vertex = queue.pop()
+        print(current_vertex)
+
+        # Get all adjacent vertices of current_vertex 
+        # If a adjacent has not been visited, then mark it
```
Comment (RavinduSachintha) [inline algorithms/gr-bfsrh/python2/bfs.py:21]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6325901)

**Comment by RavinduSachintha** [comment]
Screenshot from 2020-10-15 00-13-35
this error occurring at building action. you have to fix it and other suggestions as soon as possible.

**Pull request closed without merging**